core/frontend/src/extension/Extension.ts

```diff
@@ -32,47 +37,101 @@ export interface ExtensionManifest {
    * e.g "./lib/main.js"
    */
   readonly main: string;
-  /** List of activation events this Extension supports. */
-  readonly activationEvents: ActivationEvent[];
-}
-
-/** @alpha */
-export interface BuildExtensionManifest extends ExtensionManifest {
   /** Only valid when the Extension is loaded at build-time.
    *
    * Defines the main ES module that will be imported
    */
-  readonly module: string;
+  readonly module?: string;
+  /** List of activation events this Extension supports. */
+  readonly activationEvents: ActivationEvent[];
 }
 
-/** Describes an Extension that has already been downloaded and has a location files can be easily executed.
+/**
+ * A "ready to use" Extension (contains a manifest object and a function to execute).
+ * Will be used as the type for in-memory extensions in the ExtensionAdmin
  * @alpha
```

Comment:
Should `InstalledExtension` be a user facing concept here? It seems to be the internal detail of the ExtensionAdmin and can be moved to that file and not exported.

Reply:
Yeah definitely, its just an ExtensionAdmin concept. Will move